docs/pages/admin-guides/deploy-a-cluster/helm-deployments/argocd-helm.mdx

```diff
@@ -0,0 +1,107 @@
+
+This guide leverages ArgoCD's native Helm support to deploy the Teleport agent using the `teleport-kube-agent` Helm chart.
+
+## Prerequisites
+
+- A registered domain name. This is required for Teleport to set up TLS via
+  Let's Encrypt and for Teleport clients to verify the Proxy Service host.
+
+- An existing Kubernetes cluster you wish to provide access to via Teleport.
+
+- At least one Teleport Agent running in your cluster publicly reachable with
+  valid TLS certificates
+
+ - (!docs/pages/includes/tctl.mdx!)
+
+- An existing ArgoCD instance (version 2.10 or greater) that can deploy to the
+  above Kubernetes cluster.
+
+- The `tsh` client tool v(=teleport.version=)+ installed on your workstation.
+  You can download this from our [installation page](../../../installation/installation.mdx).
+  
+## Step 1/2. Generate a join token
+
+Teleport agents use a join token to obtain certificates and connect to Teleport. See [joining docs](../../../reference/join-methods.mdx) for more information.
+The token must be valid only on first join, the Teleport Kube agent will store its
+certificates in Kubernetes and won't need a token to join again in the future.
+In this section, we will create a token for the agent to join the Teleport cluster.
+```code
+$ tctl tokens add --type=kube,app --ttl=5m
+```
+
+You can specify the following token types:
+
+(!docs/pages/includes/token-types.mdx!)
+
+Ensure you update the values file you use for the helm chart
+to include the required roles.
+
+See the `teleport-kube-agent` [chart
+reference](../../../reference/helm-reference/teleport-kube-agent.mdx#roles) for the
+roles and token types that the chart supports.
+
+## Step 2/2. Configure and deploy the `teleport-kube-agent` Helm chart via ArgoCD
+
+1. Create a namespace for Teleport and configure its Pod Security Admission,
+   which enforces security standards on pods in the namespace:
+
+   ```code
+   $ kubectl create namespace teleport
+   namespace/teleport created
+   
+   $ kubectl label namespace teleport 'pod-security.kubernetes.io/enforce=baseline'
+   namespace/teleport labeled
+   ```
+
+2. Create a new ArgoCD application using the following as a template.
+
+``` yaml
+project: default
+source:
+  repoURL: 'https://charts.releases.teleport.dev'
+  targetRevision: (=teleport.version=)
+  helm:
+    values: |-
+
+      roles: kube,app,discovery
+      authToken: $YOUR_AUTH_TOKEN
+      proxyAddr: $YOUR_PROXY_ADDRESS
+      kubeClusterName: $YOUR_KUBE_CLUSTER_NAME
+
+      highAvailability:
+          replicaCount: 2
+          podDisruptionBudget:
+              enabled: true
+              minAvailable: 1
+  chart: teleport-kube-agent
+destination:
+  server: 'https://kubernetes.default.svc'
+  namespace: $YOUR_TARGET_NAMESPACE
```

Comment:
What's the namespace here? We're asking people to create "teleport" namespace above, should this just be "teleport"?